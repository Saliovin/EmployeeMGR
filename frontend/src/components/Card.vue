<script setup lang="ts">
interface Props {
    first_name: string;
    last_name: string;
}

const props = defineProps<Props>();
const emit = defineEmits(["view", "edit", "delete"]);
</script>

<template>
    <div class="card">
        <p>{{ `${props.first_name} ${props.last_name}` }}</p>
        <div class="modal-set">
            <button class="view" @click="emit('view')">[View]</button>
            <button class="edit" @click="emit('edit')">[Edit]</button>
            <button class="delete" @click="emit('delete')">[Delete]</button>
        </div>
    </div>
</template>

<style scoped>
button {
    padding: 0;
    margin: 0 0.2rem;
}

.card {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-color: #eeeeee;
    height: 10rem;
    width: auto;
    max-width: 13rem;
    padding: 0.5rem;
    border: 1px solid #555;
}

.modal-set {
    align-self: center;
    display: flex;
    width: 100%;
    justify-content: space-between;
}

.view {
    background-color: #9fc4e7;
}

.view:hover {
    background-color: #899fd7;
}

.edit {
    background-color: #fefe06;
}

.edit:hover {
    background-color: #d5e401;
}

.delete {
    background-color: #df6565;
}

.delete:hover {
    background-color: #c5544a;
}
</style>
