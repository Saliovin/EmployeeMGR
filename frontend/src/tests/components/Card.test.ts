import Card from '@/components/Card.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('card', () => {
    it('base', () => {
        const card = mount(Card, {
            props: {
                first_name: 'test',
                last_name: 'testing'
            }
        })
        expect(card).toMatchSnapshot()
    })
})