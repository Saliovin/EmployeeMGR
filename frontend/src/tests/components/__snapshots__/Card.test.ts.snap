// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`card > base 1`] = `
<div
  class="card"
  data-v-c6c3362a=""
>
  <p
    data-v-c6c3362a=""
  >
    test testing
  </p>
  <div
    class="modal-set"
    data-v-c6c3362a=""
  >
    <button
      class="view"
      data-v-c6c3362a=""
    >
      [View]
    </button>
    <button
      class="edit"
      data-v-c6c3362a=""
    >
      [Edit]
    </button>
    <button
      class="delete"
      data-v-c6c3362a=""
    >
      [Delete]
    </button>
  </div>
</div>
`;
